0.1.0.md

```diff
@@ -1,63 +1,141 @@
 # 0.1.0
 
+- [Fields](#fields)
+- [Encoding](#encoding)
+    - [Types](#types)
+
+
 ## Fields
 
-Field | Type | Description | Constraints | Semantics
-------|------|-------------|-------------|----------
-`spec` | `string` | The [semantic version](http://semver.org/) of the specification. | This must be a valid spec version. | -
-`domain` | `string` | The domain or namespace of the value's identity. | Allowed characters `[A-Z][a-z][0-9]-._` |  See [identity](#identity)
-`name` | `string` | The name of the value's identity. | Requires `domain` to be set. Allowed characters `[A-Z][a-z][0-9]-._` |  See [identity](#identity)
-`version` | `string` | Defines the semantic version of the identity this value is associated with. | Requires a valid SemVer version string. | See [versioning](#versioning)
-`revision` | `string` | Uniquely identifies the value for the [identity](#identity). | - | See [versioning](#versioning)
-`time` | `uint64` | A Unix epoch timestamp in nanoseconds when the message was produced. | - | -
-`value` | `<T>` | The value of the message. | The opaque value of the message. | -
-`refs` | `map[string]map[string]<T>` | A set of references to other entities. | At minimum, each `ref` must define the `name` field which is the identity being referenced. Any of the additional may be included with the exception of `refs`. | See [references](#references)
-`meta` | `map[string]<T>` | A set of metadata to annotate the entity. | - | See [metadata](#metadata)
-`labels` | `array[string]` | A set of strings that annotate the message. | Allowed characters `[A-Z][a-z][0-9]-._` | See [labels](#labels)
+- [`spec`](#spec)
+- [`time`](#time)
+- [`value`](#value)
+- [`name`](#name)
+- [`domain`](#domain)
+- [`revision`](#revision)
+- [`version`](#version)
+- [`labels`](#labels)
+- [`refs`](#refs)
+- [`meta`](#meta)
+
+### `spec`
+
+- Required: **Yes**
+- Type: string
+
+The [semantic version](http://semver.org) of the specification the fact was produced against.
+
+This is to ensure consumers can check for compatibility when consuming a fact.
+
+
+### `time`
+
+- Required: **Yes**
+- Type: integer
+
+A [Unix epoch](https://en.wikipedia.org/wiki/Unix_time) timestamp in nanoseconds.
+
+The `time` denotes when the [value](#value) was asserted and therefore valid. Subsequent facts that assert a value for the same [entity](#name) implicitly invalidates the previous values.
+
+
+### `value`
+
+- Required: **Yes**
+- Type: any type
+
+The value of an entity being asserted.
+
+A value is, by definition, immutable. If assigned to an [entity](#name), this is referred to as the *state* of the entity at the associated time.
+
+
+### `name`
+
+- Required: No
+- Type: string
+- Regexp: `^([a-z])([a-z][0-9]-._)*$`
+
+The name of the entity the [value](#value) is assigned to.
+
+The name is part of the *identity* of an entity, along with the [domain](#domain) if defined.
```

Comment:
It's not 100% clear what "if defined" refers to -- name, domain, or both. 
Maybe something like this?
"If the name is defined, the [domain](#domain) must be defined as well. Together the name and domain define the *identity* of an entity."

That's the only nitpick I had. I like this version a lot. The entity/value/state relationship is very clear, and version/revision/time is clear as well.